.github/workflows/build_and_test.yml

```diff
@@ -18,58 +20,86 @@ jobs:
       - name: Check out
         uses: actions/checkout@v2
         with:
-          fetch-depth: 0 # Fetch the whole git history, to be able to calculate some stats based on it.
+          fetch-depth: ${{ matrix.testTask != 'pythonTest' }}  # For 'pythonTest' fetch the whole git history ('false' -> '0'), to be able to calculate some stats based on it. For others, only a single commit will be fetched ('true' -> '1')
```

Comment:
Clever idea, but I'm afraid it doesn't work. For both targets, I see:

```
/usr/bin/git -c protocol.version=2 fetch --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*
```

and the condition evaluates to `fetch-depth: true` and `fetch-depth: false`. I'm afraid GH Actions cannot interpret boolean as an integer.

If we weren't in a YAML-based DSL, we could use an if expression which I don't see for GH Actions. This optimization saves ~1-2 minutes. If we want to go with it, I propose having two separate steps, with `if:` checking `pythonTest` for `fetch-depth: 0`, and `microPythonTest` for `fetch-depth: 1`. However, this comes at a cost of making the workflow more convoluted. I personally wouldn't pursue such optimization, but I'm not against - leaving it to you :)

Reply:
Good catch, will recheck.

> I'm afraid GH Actions cannot interpret boolean as an integer.

Well, I haven't just made it up, I've seen it in GH Docs: https://docs.github.com/en/actions/learn-github-actions/expressions#operators

![image](https://user-images.githubusercontent.com/20105593/145581794-e958febb-4448-49bc-bcea-e5789d0e5f02.png)

However, maybe it's about something else...

Reply:
Note: this is just optimization, nothing functional. We don't have to block on it :)

Reply:
Turns out the history is needed for the `generate-box-tests-reports.main.kts` script when it builds the plot. If there is no history, it crashes. So let me leave the logic of always fetching the whole history